### PR TITLE
Notifications for Node and Go versions not on image

### PR DESCRIPTION
**Pull request by shilovmaksim**
Current pull request contains following changes:
- Adding notification for new versions of Node and Go not on image.
- Slight rework and optimization.

#### Run example: 
https://github.com/shilovmaksim/versions-package-tools/actions/runs/1548396937

#### Message example: 
image

### CODE REVIEW

File: .github/workflows/get-tools-new-versions.yml [hunk 1]
```diff
@@ -16,17 +16,40 @@ jobs:
...
+            filter-parameter: 'version'
+            filter-arch: 'x64'
+          - name: 'Go'
+            image: 'https://avatars.githubusercontent.com/u/4314092?s=200&v=4'
+            releases-url: 'https://raw.githubusercontent.com/actions/go-versions/main/versions-manifest.json'
+            filter-parameter: 'version'
+            filter-arch: 'x64'
     name: 'Searching for new versions of ${{ matrix.tool.name }}'
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Comment (al-cheb) [inline .github/workflows/get-tools-new-versions.yml:43]: What is the reason to use windows?

File: .github/workflows/get-tools-new-versions.yml [hunk 2]
```diff
@@ -16,17 +16,40 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
       - id: get-new-tool-versions
         name: Get new tool versions
-        run: echo "::set-output name=versions-output::$(./get-new-tool-versions/verify-new-tool-version-added-to-image.ps1 -ToolName ${{ matrix.tool.name }})"
+        run: echo "::set-output name=versions-output::$(./get-new-tool-versions/verify-new-tool-version-added-to-image.ps1 `
```
Comment (al-cheb) [inline .github/workflows/get-tools-new-versions.yml:48]: Let's create an additional variable for versions-output.

File: get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1 [hunk 3]
```diff
@@ -0,0 +1,25 @@
+function Search-ToolsVersionsNotOnImage {
+    param (
+        [string]$ToolName,
+        [string]$ReleasesUrl,
+        [string]$FilterParameter,
+        [string]$FilterArch
+    )
+    
+    $stableReleases = (Invoke-RestMethod $ReleasesUrl) | Where-Object stable -eq $true
```
Comment (al-cheb) [inline get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1:9]: Why do we need ()?

File: get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1 [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    if ($ToolName -eq "Node") {
+        $stableReleaseVersions = $stableReleases | ForEach-Object {$_.$FilterParameter.split(".")[0] + ".0"} |
+            Select-Object -Unique
+    } else {
+        $stableReleaseVersions = $stableReleases | ForEach-Object {$_.$FilterParameter.split(".")[0,1] -join"."} |
+            Select-Object -Unique
+    }
+    $toolsetUrl = "https://raw.githubusercontent.com/actions/virtual-environments/main/images/win/toolsets/toolset-2022.json"
+    $latestVersion = ((Invoke-RestMethod $toolsetUrl).toolcache |
```
Comment (al-cheb) [inline get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1:18]: Let's replace $latestVersion -> $latestMinorVesion , remove () and use Foreach to $_.versions.Replace("*","0").

File: .github/workflows/get-tools-new-versions.yml (hunk 1 above)
Comment (shilovmaksim) [reply .github/workflows/get-tools-new-versions.yml:43]: Previously it was changed to get rid of "Where-Object: Cannot convert value "16" to type "System.Version". Error: "Version string portion was too short or too long." with Node sorting. It worked locally with PS 7, but was failing in the workflow.
Currently it just left with windows after all the debugging.

File: get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1 (hunk 3 above)
Comment (shilovmaksim) [reply get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1:9]: For some reason where-object sorting not work with invoke-restmethod without parentheses:
```
PS D:\Repo\Temp\versions-package-tools> (Invoke-RestMethod $ReleasesUrl | Where-Object stable -eq $true).count
0
PS D:\Repo\Temp\versions-package-tools> ((Invoke-RestMethod $ReleasesUrl) | Where-Object stable -eq $true).count
183
```

File: .github/workflows/get-tools-new-versions.yml (hunk 1 above)
Comment (al-cheb) [reply .github/workflows/get-tools-new-versions.yml:43]: Where at that line do we get 16?

Comment (shilovmaksim) [reply .github/workflows/get-tools-new-versions.yml:43]: Cached "node" versions are specified in the toolset with just major version opposed to other tools with minor versions. So in the line 21 of "`verify-common-tool-parser.psm1`" we will get just 16 with `$latestVersion.TrimEnd(".*")` which causing error in comparison.

Comment (al-cheb) [reply .github/workflows/get-tools-new-versions.yml:43]: Now we don't have this statement. I think we can return ubuntu.

File: .github/workflows/get-tools-new-versions.yml (hunk 2 above)
Comment (shilovmaksim) [reply .github/workflows/get-tools-new-versions.yml:48]: Ok

File: .github/workflows/get-tools-new-versions.yml (hunk 1 above)
Comment (shilovmaksim) [reply .github/workflows/get-tools-new-versions.yml:43]: Indeed

Comment (shilovmaksim) [reply .github/workflows/get-tools-new-versions.yml]: Updated

File: get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1 (hunk 4 above)
Comment (shilovmaksim) [reply get-new-tool-versions/parsers/verify-added-to-image/verify-common-tool-parser.psm1]: Updated

**Review by al-cheb — APPROVED** [review]

**Review by miketimofeev — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**